tools/link.py

```diff
@@ -220,19 +220,32 @@ def get_js_sym_info():
     input_files.append(read_file(jslib))
   content = '\n'.join(input_files)
   content_hash = hashlib.sha1(content.encode('utf-8')).hexdigest()
+  library_syms = None
+
+  cache_file = f'symbol_lists/{content_hash}.json'
+  # When we are racing with other processes the existence of the file
+  # at this point in time only means its likely a cache hit, since another
+  # process could be performing a cache cleanup.
```

Comment:
I don't follow the sentence after `, since`. Is the idea "if the file exists now, we will likely still have it later, but we do need to be careful of a cache clearing in the middle" - ?

Reply:
If so, that sentence might work, plus "but if the file does not exist now, a cache miss is likely (unless we get lucky and another process generates it), and in that case we generate the info now rather than later, as it is more efficient to do so now before we take the cache lock".

Reply:
Yes, exactly there could be another process performing a cache cleanup between our time-of-check and time-of-use of the file.